### PR TITLE
Add cluster proxy addon

### PR DESCRIPTION
**Pull request by JakobGray**
Issue: https://github.com/open-cluster-management/backlog/issues/15166

Signed-off-by: Jakob Gray <20209054+JakobGray@users.noreply.github.com>

### CODE REVIEW

File: controllers/common_test.go [hunk 1]
```diff
@@ -292,6 +292,12 @@ func kind(kind string) schema.GroupKind {
 // 			Result:       nil,
 // 		},
 // 		{
+// 			Name:         "Test: ensureSubscription - Cluster Proxy Addon",
```
Comment (zkayyali812) [inline controllers/common_test.go:295]: Did you mean to leave this commented out?

Comment (JakobGray) [reply controllers/common_test.go:295]: Yes, that entire file is commented out so I added the entry without changing that

**Review by zkayyali812 — APPROVED** [review]

**Pull request merged**